zowe/zlux-app-manager
# Load recognizers & actions upon Refresh Applications

Load recognizers & actions upon calling install-app.sh or adding a plugin

**To test:**

Put something like this in /appDir/config/recognizers OR /actions, titled by 'app identifier' 

So for example, in `zlux\zlux-editor\config\recognizers`
I may have a file like, `org.zowe.editor`
with contents,
```
{
  "recognizers":[
    {
      "id":"com.rs.iuw.mq.openapp.forward",
      "capabilities": [
        "com.ibm.iuw.extension.menu"
      ],
      "clause": {
        "op": "AND",
        "args": [
         {"prop":["primaryApplicationName","KMQ"]},
         {"prop":["primaryTableName","QMMSG_SUM"]}
        ]
      }
    }
  ]
}
```

Previous dispatcher & action loading code didn't actually work due to a timing issue with "forEach" even if you manually added things into the config location. If you run my code now, in the log you will see something like
2020-07-06 04:28:58.156 <ZWED:> ts6321 INFO (org.zowe.zlux.ng2desktop,:) ZWED5055I - Loaded 1 recognizers for App(org.zowe.editor)
